syntax = "proto2";
package spy;

message LoginInfo{
	required string nickname = 1;
	required string wechatid = 2;
	required string wxid = 3;
	required string phone = 4;
	required string profilephoto = 5;
}

message Contact{
	required string wxid = 1;
	optional string nickname = 2;
	optional string wechatid = 3;
	optional string remark = 4;
	optional string profilephoto = 5;
	optional string profilephoto_hd = 6;
	optional int64 sex = 7;
	optional string whats_up = 8;
	optional string country = 9;
	optional string province = 10;
	optional string city = 11;
	optional int64 source = 12;
	optional int64 status = 13;
}

message ContactList{
	repeated Contact contact = 1;
}

message ChatroomMemberList{
	required string wxid = 1;
	repeated Contact contact = 2;
}

message Message{
	required int64 type = 1;
	required int64 self = 2;
	optional string head = 3;
	required string wxid1 = 4;
	optional string wxid2 = 5;
	required string content = 6;
}

message MessageList{
	repeated Message message = 1;
}

message Qrcode{
	required int64 success = 1;
	required string qrcode = 2;
}

message Response{
	required int64 type = 1;
	required int64 pid = 2;
	optional string uuid =3;
	optional LoginInfo login_info= 4;
	optional ContactList contact_list= 5;
	optional ChatroomMemberList chatroom_member_list = 6;
	optional MessageList message_list = 7;
	optional Qrcode qrcode = 8;
	optional string debug = 9;
	optional string info = 10;
	optional string warning = 11;
	optional string error = 12;
}

message Request{
	required int64 cmd = 1;
	required string uuid = 2;
	optional string wxid1 = 3;
	optional string chatroom_wxid = 4;
	optional string at_wxid = 5;
	optional string content = 6;
	optional string encryptusername = 7;
	optional string ticket = 8;
	optional int64 update = 9;
}